### PR TITLE
DOC-4528 async hash examples

### PR DESCRIPTION
**Pull request by andy-stark-redis**
Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/lettuce-io/lettuce-core/blob/main/.github/CONTRIBUTING.md).
- [x] You submit test cases (unit or integration tests) that back your changes.

 
My first attempt at a Lettuce async code example for the [hash datatype page](https://redis.io/docs/latest/develop/data-types/hashes/). Although it passes the CI, I expect it's not ideal in terms of good practice. In particular:

- Is it OK to run the `del()` call at the start from the `CompletableFuture.allOf()` call?
- Where an example snippet depends on the results of the previous snippet, I've taken the variable from the previous snippet and continued from it (eg, `CompletableFuture<Void> hmGet = setGetAll.thenCompose(...` Is this the best way to handle this?

Any other feedback about style, technique, and general Java knowledge would be very welcome :-)

### CODE REVIEW

File: src/test/java/io/redis/examples/async/HashExample.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                        assertThat(res.toString()).isEqualTo("[KeyValue[crashes, 1], KeyValue[owners, 1]]");
+                        return res;
+                    })
+                    // REMOVE_END
+                    .thenAccept(System.out::println)
+                    // >>> [KeyValue[crashes, 1], KeyValue[owners, 1]]
+                    .toCompletableFuture();
+            // STEP_END
+
+            CompletableFuture.allOf(
```
Comment (tishun) [inline src/test/java/io/redis/examples/async/HashExample.java:156]: My bad, deleted my previous comment.

Why only add `hIncrBy` and `incrByGetMget` ? Perhaps we need to add all the completable futures??

**Review by tishun — APPROVED** [review]
LGTM

File: src/test/java/io/redis/examples/async/HashExample.java (hunk 1 above)
Comment (tishun) [reply src/test/java/io/redis/examples/async/HashExample.java:156]: My bad, they are all used as basis for the other completable futures, thanks for clarifying offline.

**Review by ggivo — COMMENTED** [review]
LGTM

**Pull request merged**